src/main/kotlin/org/rust/lang/core/resolve/NameResolution.kt

```diff
@@ -216,7 +216,11 @@ fun processPathResolveVariants(lookup: ImplLookup, path: RsPath, isCompletion: B
             // then resolving associated type to <Self as Tr>::Item, and then substituting `{Self => T}` into it
 
             for ((element, subst) in lookup.findImplsAndTraits(TyTypeParameter.named(base))) {
```

Comment:
Does it make sense to fold this for loop with the previous if, which handles functions?

Reply:
Probably yes, we need the same "inherent wins" logic for constants.